comp/core/hostname/mock.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package hostname
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+)
+
+type mockService struct {
+	name string
+}
+
+var _ Mock = (*mockService)(nil)
+
+func (m *mockService) Get(ctx context.Context) (string, error) {
+	return m.name, nil
+}
+
+func (m *mockService) GetSafe(ctx context.Context) string {
+	return m.name
+}
+
+func (m *mockService) Set(name string) {
```

Comment:
Exposing a `Set` method is convenient but can lead to inconsistency because some parts of the code could already call `Get` before having a chance to call this function. Do you think it is relevant to remove this function?

Reply:
Good point. I don't think there's any need for it - the real hostname shouldn't ever change while the agent is running, so I don't know why a test would need to change it itself. I've removed this (which also means that we don't need a separate `Mock` interface anymore).